### PR TITLE
Add JWT authentication

### LINKED ISSUE #48: Add Backend User Authentication
User accounts (so they can access/manage their old links)
- [X] (b) user schema
- [X]  (b) probably use passport with jwt/google account/other account login strategies
- [X]  (b) login endpoint
- [X]  (b) register endpoint

### PR DESCRIPTION
**Pull request by Ash1eyC0des**
## Summary
closes #48
Added authentication with Passport-JWT: users/signup & users/login routes
## Checklist
- [X] Tested signup & login routes in Postman

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -1,13 +1,17 @@
 const express = require('express');
 const mongoose = require('mongoose');
-require('dotenv').config();
+const passport = require('passport');
+require('dotenv').config({ path: './src/config/.env' });
```
Comment (cherylli) [inline server/index.js:4]: this line gives me error. Let's use the default .env location which is the root of server :)

**Review by cherylli — COMMENTED** [review]
image
with login, i'm not getting proper response for wrong password

would be good if the api call response can be consistent , if you look at http://localhost:4002/api/docs/
for existing api endpoints

everything else look good!

**Review by harshilsharmaa — APPROVED** [review]

**Comment by cherylli** [comment]
the delete function is gone, maybe during resolve conflicts
https://github.com/webdev-20/TLL-hacktoberfest-2022/blob/65129446b807b26433e0914a41be60fb32894dce/server/src/controllers/links.controller.js

image

File: server/index.js (hunk 1 above)
Comment (Ash1eyC0des) [reply server/index.js]: Done :)

Comment (Ash1eyC0des) [reply server/index.js]: I also added some validation to the signup route and re-tested everything.

**Review by cherylli — APPROVED** [review]
looks good!

**Pull request merged**